src/unix/lwt_io.ml

```diff
@@ -1454,6 +1454,41 @@ let with_file ?buffer ?flags ?perm ~mode filename f =
     (fun () -> f ic)
     (fun () -> close ic)
 
+let open_temp_file_with_filename:
+  ?buffer : Lwt_bytes.t ->
+  ?perm : Unix.file_perm ->
+  unit -> (string * output_channel) Lwt.t
+  = fun ?buffer ?perm ->
+    let f = Some [Unix.O_WRONLY; Unix.O_CREAT; Unix.O_TRUNC;
+                  Unix.O_NONBLOCK; Unix.O_EXCL] in
+    let rec helper n =
+      let fname = "lwt_io_temp_file_" ^ string_of_int n in
```

Comment:
In case of a process rapidly opening temporary file names in a loop, and keeping them open for a while, this will cause the first `N` filenames to be already taken on iteration `N + 1`, and this function will perform `N` system calls that will fail. I recommend using a random number generator instead, see [`Random.State`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Random.State.html). There is also an [example](https://github.com/ocaml/ocaml/blob/7a315bdb5e66bffdd7cef536373a2e0b753d1b7f/stdlib/filename.ml#L233) of it being used in the stdlib.

Reply:
I would add that the prefix to the file name shouldn't be hardcoded and should be passed into the function so the user can give it a descriptive name.